src/inlay-hints.ts

```diff
@@ -0,0 +1,233 @@
+// This file implements the client side of the proposed inlay hints
+// extension to LSP. The proposal is based on the one at
+// https://github.com/microsoft/language-server-protocol/issues/956,
+// with some modifications that reflect discussions in that issue.
+// The feature allows the server to provide the client with inline
+// annotations to display for e.g. parameter names at call sites.
+// The client-side implementation is adapted from rust-analyzer's.
+
+import * as vscode from 'vscode';
+import * as vscodelc from 'vscode-languageclient/node';
+
+import {ClangdContext, isClangdDocument} from './clangd-context';
+
+export function activate(context: ClangdContext) {
+  const feature = new InlayHintsFeature(context);
+  context.client.registerFeature(feature);
+}
+
+// Currently, only one hint kind (parameter hints) are supported,
+// but others (e.g. type hints) may be added in the future.
+enum InlayHintKind {
+  Parameter = 'parameter'
+}
+
+interface InlayHint {
+  range: vscodelc.Range;
+  kind: InlayHintKind;
```

Comment:
I think this implies the server should never send other values for Kind, I'd probably type this as InlayHintKind|string instead, up to you.